### PR TITLE
Add tweaks to fix failed 0.18 tests

### PR DESCRIPTION
**Pull request by jeanconn**
Add tweaks to fix failed 0.18 tests

Includes reverted numexpr 2.2.2 to deal with 'copy_args' problems from hd5 queries.
Includes fontconfig/matplotlib updates for gtk to work for gui_fit.py

### CODE REVIEW

**Comment by jeanconn** [comment]
@taldcroft 

These are just some ideas.  My last full clean build on unagi with this branch did provide a working matplotlib gtk backend that seemed functional with guit_fit.py.

**Comment by jeanconn** [comment]
Nevermind.  This looks broken again (tested off last non-clean build instead of the clean build I intended to test).

File: cfg/python_modules.cfg [hunk 1]
```diff
@@ -29,17 +29,17 @@ modules:
...
-  #     test    : |
-  #       test -e .installed
-  #       cd $prefix
-  #       ${prefix_arch}/bin/python -c "import ${module}"
-  #       ${prefix_arch}/bin/python -c "import pylab"
-  #     install : |
-  #       export PKG_CONFIG_PATH=${prefix_arch}/lib/pkgconfig
-  #       ${prefix_arch}/bin/python setup.py install
-  #       touch .installed
+  - name : matplotlib
```
Comment (jeanconn) [inline cfg/python_modules.cfg]: Ah.  I forgot.  If we want to build matplotlib from source and pick up pygtk and gtk libs, those libs need to be build before matplotlib, which is I suppose what we'd get if we set

```
python_libs_64: xtime git
# gtk pygtk
```

back to

```
python_libs_64: xtime git gtk pygtk
```

File: cfg/anaconda-32.cfg [hunk 2]
```diff
@@ -9,8 +9,11 @@ cmds :
     cd $prefix
     ${prefix_arch}/bin/python -V
   build   : |
-    bash ${pkg_dir}/Miniconda-latest-Linux-x86.sh -b -p ${prefix_arch}
+    bash ${pkg_dir}/Miniconda-latest-Linux-x86-jean.sh -b -p ${prefix_arch}
```
Comment (taldcroft) [inline cfg/anaconda-32.cfg]: ???

**Comment by taldcroft** [comment]
Sorry to have dumped this on you without having had a chance to sit down together and discuss.  You are coming across various breadcrumbs in my unsuccessful path to getting something working.  BTW, testing Tk would also be good.

File: cfg/anaconda-32.cfg (hunk 2 above)
Comment (jeanconn) [inline cfg/anaconda-32.cfg]: Sorry, just noted this in the commit comments.  Your Miniconda...sh had read only for you permissions so I couldn't even run it.

**Comment by taldcroft** [comment]
(Including maddening things like failing to build git because of a conflict with the conda Tk library, hence the bizarre removing tk, build git, then re-install tk.  Speaking of which I never tested if the git GUI tools actually worked.  I suppose if we there was a recent-enough git on chimchim then we could punt on bundling git with Ska.

**Comment by taldcroft** [comment]
Copy on the miniconda problem.  I thought you have somehow patched that.

**Comment by jeanconn** [comment]
(And I meant to just ask you about fixing the perms on Miniconda, not just including the commit)

**Comment by jeanconn** [comment]
Regarding Tk tests, do you have ideas?  Looks like aca_movie is working which I suppose is testing its Perl->Tk stuff (and I did call this with perl aca_movie.pl to avoid getting perlska from the aca_movie shebang).

**Comment by jeanconn** [comment]
Meanwhile, this isn't looking so hot

```
fido: gitk
Application initialization failed: Can't find a usable init.tcl in the following directories: 
    /opt/anaconda1anaconda2anaconda3/lib/tcl8.5 /usr/lib/tcl8.5 /lib/tcl8.5 /usr/library /library /tcl8.5.18/library /tcl8.5.18/library
This probably means that Tcl wasn't installed properly.
```

**Comment by taldcroft** [comment]
```
$ ipython --pylab=tk
>>> plot()
$ gitk
```

**Comment by jeanconn** [comment]
Yeah, no joy on the tk backend either.  I think conda tk is broken.

**Comment by taldcroft** [comment]
I think that our mucking about has broken conda tk.  I bet if you use a clean conda env that Tk will work.

**Comment by jeanconn** [comment]
Actually, I just suck at testing.  Now that my matplotlib has finished rebuilding, --pylab=tk, --pylab=qt, and --pylab=gtk are all working.

**Comment by taldcroft** [comment]
This kind of testing is maddening to get right.

**Comment by jeanconn** [comment]
Does that just leave these as open tasks on this fix-up thread?

 - [ ] rebuild clean env with new order to confirm gtk install process stability 
 - [ ] fix git gui tools
 - [ ] fix starcheck Inline problem

**Comment by taldcroft** [comment]
But `gitk` is still failing?  We can live without bundling git.  Or drop the gitk support, which I never use now anyway.

**Comment by jeanconn** [comment]
Yes, I don't know if gitk might need a hint on where to find that init.tcl (environment variable?).

**Comment by taldcroft** [comment]
Once you think you have a working build setup then you can build in `/proj/sot/ska/dev` and test that (32-bit and 64-bit), and finally merge this into `skare-0.18`.  I just moved `dev` to `dev-aldcroft` so you should be clear for a clean install (including various command line tools that get make-installed from starcheck, timelines, cmd_states etc).

**Comment by jeanconn** [comment]
OK.​  Heading outside today though, so probably won't make any more
progress until tomorrow (rebuilding 64 bit again on the side now).

**Comment by taldcroft** [comment]
And when everything looks good on our side, go to greta and move the existing `dev` aside and install your new one.  As discussed with Mark B the deadline on that is Friday after next.

**Comment by taldcroft** [comment]
Good plan to get away from console!

**Comment by jeanconn** [comment]
Did you have another post-install summary from 0.17 handy?  I think the current list is

- mica
   - python setup.py install # do we need to make install the scripts for anything in /dev or GRETA?
- starcheck
   - make install & python setup.py install
- cmd_states
   - make install # for scripts but not python setup.py because already in skare
- timelines?  # does this need to be installed or can it just be locally tested?

**Comment by jeanconn** [comment]
This build passes my tests on 32 and 64 bit builds:

 - [x] gitk
 - [x] ipython --pylab=qt
 - [x] ipython --pylab=gtk
 - [x] starcheck: make test
 - [x] eng_archive: Ska.engarchive.test()

64 bit passes

 - [x] xija: gui_fit.py

Issues:

Not sure how to run any kadi tests without installing kadi events db.

The 32 bit version does not have Sybase support.  I think this is basically OK as we don't have Sybase access on GRETA and we don't need to run starcheck on the HEAD LAN from 32 bit machines.

Miniconda-latest-Linux-x86-jean.sh appears to be 3.10.1 and, now that I can read Miniconda-latest-Linux-x86.sh, I see it is 3.9.1.   Do we have any problems upgrading (now that I've don't this prelim testing against 3.10.1)?

I'm confused about pushing this to GRETA dev while Mark is testing against that already.

File: cfg/anaconda-32.cfg (hunk 2 above)
Comment (taldcroft) [inline cfg/anaconda-32.cfg]: I just removed the original Miniconda-latest-Linux-x86.sh so you can `mv` your version to that name and fix group/perms.  Then revert this change.

**Comment by taldcroft** [comment]
Awesome!  Pending the one fix about Miniconda..-jean.sh you can merge this.  I don't think there is a problem with the Miniconda version.

**Comment by jeanconn** [comment]
Ok.  I rebased without the Miniconda-jean bit (I didn't add anything helpful to the git history).

**Comment by jeanconn** [comment]
​OK.  Merged.  I think the only slight issue with the updated Miniconda
version is that for the 32bit install it now has to downgrade python's
build version when the install completes from pkgs.conda.
